hi-ml-histopathology/testhisto/testhisto/utils/test_output_utils.py

```diff
@@ -214,15 +217,34 @@ def test_gather_results_distributed() -> None:
     run_distributed(test_gather_results, world_size=2)
 
 
-def test_collate_results() -> None:
-    epoch_results = _create_epoch_results(batch_size=3, num_batches=5, rank=0, device='cpu')
+def _test_collate_results(epoch_results: EpochResultsType, total_num_samples: int) -> None:
+    collated_results = collate_results_on_cpu(epoch_results)
 
-    collated_results = collate_results(epoch_results)
+    for key, epoch_elements in collated_results.items():
+        expected_elements = [batch_results[key] for batch_results in epoch_results]
+        if key != ResultsKey.LOSS:  # loss is a single tensor per batch
+            assert len(epoch_elements) == total_num_samples
+            # Concatenated lists:
+            expected_elements = [elem for batch_elements in expected_elements for elem in batch_elements]
 
-    for key, value in collated_results.items():
-        epoch_values = [batch_results[key] for batch_results in epoch_results]
-        if key == ResultsKey.LOSS:
-            assert value == epoch_values
-        else:
-            expected: List = sum(epoch_values, [])  # concatenated lists
-            assert_close(value, expected)
+        for elem in epoch_elements:
+            if isinstance(elem, torch.Tensor):
+                assert not elem.is_cuda, f"{key} tensor must be on CPU: {elem}"
+        assert_close(epoch_elements, expected_elements, check_device=False)
+
+
+def test_collate_results_cpu() -> None:
+    num_batches = 5
+    batch_size = 3
+    epoch_results = _create_epoch_results(batch_size, num_batches, rank=0, device='cpu')
+    _test_collate_results(epoch_results, total_num_samples=num_batches * batch_size)
+
+
+@pytest.mark.skipif(torch.cuda.device_count() < 2, reason="Not enough GPUs available")
+@pytest.mark.gpu
+def test_collate_results_multigpu() -> None:
+    num_batches = 5
+    batch_size = 3
+    epoch_results = _create_epoch_results(batch_size, num_batches, rank=0, device='cuda:0') \
+        + _create_epoch_results(batch_size, num_batches, rank=1, device='cuda:1')
```

Comment:
Can we do this in three lines to avoid `\`?

Reply:
Is it bad enough to block the PR? 🙃